python/ml4ir/applications/classification/pipeline.py

```diff
@@ -0,0 +1,131 @@
+import sys
+from argparse import Namespace
+
+from tensorflow.keras.metrics import Metric, Precision
+from tensorflow.keras.optimizers import Optimizer
+
+from ml4ir.applications.classification.config.parse_args import get_args
+from ml4ir.applications.classification.model.losses import categorical_cross_entropy
+from ml4ir.applications.ranking.model.metrics import metric_factory
+from ml4ir.base.config.keys import TFRecordTypeKey
+from ml4ir.base.data.relevance_dataset import RelevanceDataset
+from ml4ir.base.features.preprocessing import get_one_hot_vectorizer
+from ml4ir.base.model.losses.loss_base import RelevanceLossBase
+from ml4ir.base.model.optimizer import get_optimizer
+from ml4ir.base.model.relevance_model import RelevanceModel
+from ml4ir.base.model.scoring.scoring_model import ScorerBase, RelevanceScorer
+from ml4ir.base.model.scoring.interaction_model import InteractionModel, UnivariateInteractionModel
+from ml4ir.base.pipeline import RelevancePipeline
+
+from typing import Union, List, Type
+
+
+class ClassificationPipeline(RelevancePipeline):
+    """
+    Pipeline defining the classification models.
+    """
+    def __init__(self, args: Namespace):
+        self.loss_key = args.loss_key
+        super().__init__(args)
+
+    def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
+        """
+        Creates RelevanceModel suited for classification use-case.
+
+        NOTE: Override this method to create custom loss, scorer, model objects.
+        """
+
+        # Define interaction model
+        interaction_model: InteractionModel = UnivariateInteractionModel(
+            feature_config=self.feature_config,
+            feature_layer_keys_to_fns=feature_layer_keys_to_fns,
+            tfrecord_type=TFRecordTypeKey.EXAMPLE,
```

Comment:
@Ullimague Might be good to not hard code tfrecord_type, no?

Reply:
You can set it as a default argument.

Reply:
Good catch, it's already set as default actually, we should use self.tfrecord_type here.